avm/src/main.rs

```diff
@@ -46,10 +48,26 @@ fn parse_version(version: &str) -> Result<Version, Error> {
         Version::parse(version).map_err(|e| anyhow::anyhow!(e))
     }
 }
+
+fn parse_install_target(version_or_commit: &str) -> Result<InstallTarget, Error> {
+    parse_version(version_or_commit)
+        .map(InstallTarget::Version)
+        .or_else(|version_error| {
+            avm::check_and_get_full_commit(version_or_commit)
+                .map(InstallTarget::Commit)
+                .map_err(|commit_error| {
+                    anyhow!("Not a valid version or commit: {version_error}, {commit_error}")
+                })
+        })
+}
+
 pub fn entry(opts: Cli) -> Result<()> {
     match opts.command {
         Commands::Use { version } => avm::use_version(version),
```

Comment:
I've tried it and confirm it works as expected but one area that might need improvement is `avm use` command as it requires putting the full version-hash `avm use 0.28.0-6cf200493a307c01487c7b492b4893e0d6f6cb23` and doesn't work with `avm use 6cf200`.